### PR TITLE
Refactor(eos_designs): Wildcard dict to list management-settings local-users

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

local-users

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: (Claus)
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [X] Verify that CI pass

- Reviewer 2: (Guillaume)
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ Gives ability to monitor and react to Syslog messages provides a powerful and fl
 
 ```yaml
 event_handlers:
-  < event_handler_name >:
+  - name: < event_handler_name >
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md]: This change is technically not related to the PR (local-users) and there is no code in the PR modifying the behavior - is it expected?

File: ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md [hunk 2]
```diff
@@ -95,7 +95,7 @@ event_handlers:
 
 ```yaml
 event_handlers:
-  evpn-blacklist-recovery:
+  - name: evpn-blacklist-recovery
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md]: Same comment as the one above.

File: ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md (hunk 1 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md:85]: its not in code, but I found it with a variable key so updated it.

Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md]: its by mistake, I am creating a separate PR for event_handlers.

File: ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md (hunk 2 above)
Comment (Shivani-chourasiya) [reply ansible_collections/arista/avd/roles/eos_designs/doc/management-settings-v4.0.md]: removing it from this PR

**Review by gmuloc — APPROVED** [review]
LGTM
